src/core/src/op/paged_attention.cpp

```diff
@@ -146,7 +146,26 @@ void PagedAttentionExtension::validate_and_infer_types() {
                           get_input_element_type(12),
                           ".");
 
-    set_output_type(0, get_input_element_type(0), get_input_partial_shape(0));
+    // value head_size may be not same with key
+    auto out_ps = get_input_partial_shape(0);
+    const auto& key_ps = get_input_partial_shape(1);
+    const auto& value_ps = get_input_partial_shape(2);
+    if (out_ps.rank().is_static()) {
+        out_ps[1] = Dimension::dynamic();
+        if (key_ps.rank().is_static() && value_ps.rank().is_static() && out_ps[1].is_static() &&
```

Comment:
The `out_ps[1]` is set to dynamic line above will never  be static in this check.

the checks ` key_ps[1].is_static() && value_ps[1].is_static()` could be avoided, as ov::Dimension supports operators `*` and `/`.

```cpp
out_ps[1] *= value_ps[1] / key_ps[1];
```
If value and keys are intervals the result also will be interval, but there is a risk the operator `/` will give empty interval (invalid) and division result should be asserted e.g by this [function](https://github.com/openvinotoolkit/openvino/blob/faa3ab2bf066ce6a7da7e1fff7e5c1e2a88e1aa9/src/core/shape_inference/include/dimension_util.hpp#L230).

Reply:
Done.